include/envoy/network/connection.h

```diff
@@ -294,6 +339,13 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    *         occurred an empty string is returned.
    */
   virtual absl::string_view transportFailureReason() const PURE;
+  DynamicSocketOptionsPtr dynamic_socket_options_ptr_;
```

Comment:
I realized that this needs to be private to avoid, unintended access. I will update this on the next patch-set once I understand the test failure.